drivers/hmis/app/jobs/hmis/match_candidates_job.rb

```diff
@@ -0,0 +1,43 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Hmis
+  class MatchCandidatesJob < BaseJob
+    include NotifierConfig
+
+    queue_as ENV.fetch('DJ_LONG_QUEUE_NAME', :long_running)
+
+    # @param opportunity_ids [Array] an array of opportunity IDs to build candidate pools for.
+    # If nil, builds pools for all active Opportunities
+    # @param backoff_time [ActiveSupport::Duration] if provided, the job will _not_ re-run the match engine for
+    # candidate pools that have had candidates generated within the backoff_time.
+    def perform(opportunity_ids: nil, backoff_time: nil)
+      opportunities = opportunity_ids ? Hmis::Ce::Opportunity.where(id: opportunity_ids) : Hmis::Ce::Opportunity.active
+      log("Building candidate pools for #{opportunities.count} opportunities")
+      Hmis::Ce::Match::CandidatePoolBuilder.new(opportunities).perform
+
+      # Find candidate pools for these opportunities
+      candidate_pool_ids = opportunities.reload.pluck(:candidate_pool_id).compact.uniq
+      candidate_pool_scope = Hmis::Ce::Match::CandidatePool.where(id: candidate_pool_ids)
+      # If backoff_time was provided, only call the match engine for candidate pools that have not been updated since backoff_time
+      cp_t = Hmis::Ce::Match::CandidatePool.arel_table
+      candidate_pool_scope = candidate_pool_scope.where(cp_t[:candidates_generated_at].lt(Time.current - backoff_time)) if backoff_time
```

Comment:
could add a scope candidate_pool_ids so you can do 
```
Hmis::Ce::Match::CandidatePool.where(id: candidate_pool_ids).candidates_generated_before(datetime)
```

Reply:
I ended up removing this since the `mark_units_available.rb` procedure to check for opportunities that are "stale" was using similar logic, and I didn't think we needed it in both places. Also, I wasn't sure how to incorporate the scope you're suggesting on CandidatePool with the arel logic on Opportunity, but let me know if I should pursue that further